build/Linux+BSD/portable/bintray.sh

```diff
@@ -0,0 +1,272 @@
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+trap 'exit 1' ERR
+
+which curl || exit 1
+which bsdtar || exit 1 # https://github.com/libarchive/libarchive/wiki/ManPageBsdtar1 ; isoinfo cannot read zisofs
+which grep || exit 1
+which zsyncmake || exit 1
+
+BINTRAY_USER=$BINTRAY_USER # env
+BINTRAY_API_KEY=$BINTRAY_API_KEY # env
+
+BINTRAY_REPO_OWNER="musescore"
+WEBSITE_URL="http://musescore.org"
+ISSUE_TRACKER_URL="https://musescore.org/project/issues"
+VCS_URL="https://github.com/musescore/MuseScore.git" # Mandatory for packages in free Bintray repos
+LICENSE="GPL-2.0"
+
+API="https://api.bintray.com"
+FILE="$1"
+
+[ -f "$FILE" ] || { echo "Please provide a valid path to a file" >&2 ; exit 1 ;}
+
+# GENERIC NAMING SCHEME:
+# File:    <app_name>-<version>-<arch>.AppImage
+# Package: <app_name>-<arch>
+#
+# MUSESCORE NAMING SCHEME:
+# File:    MuseScore-X.Y.Z-<arch>.AppImage (e.g. MuseScore-2.0.3-x86_64)
+# Package: MuseScore-<arch> (e.g. MuseScore-x86_64)
+#
+# NIGHTLY NAMING SCHEME:
+# File:    MuseScoreNightly-<datetime>-<branch>-<commit>-<arch>.AppImage
+#    (e.g. MuseScoreNightly-201601151332-master-f53w6dg-x86_64.AppImage)
+# Package: MuseScoreNightly-<branch>-<arch> (e.g. MuseScoreNightly-master-x86_64)
+
+# Read app name from file name (get characters before first dash)
+APPNAME="$(basename "$FILE" | sed -r 's|^([^-]*)-.*$|\1|')"
+
+# Read version from the file name (get characters between first and last dash)
+VERSION="$(basename "$FILE" | sed -r 's|^[^-]*-(.*)-[^-]*$|\1|')"
+
+# Read architecture from file name (characters between last dash and .AppImage)
+ARCH="$(basename "$FILE" | sed -r 's|^.*-([^-]*)\.AppImage$|\1|')"
+
+case "${ARCH}" in
+  x86_64|amd64 )
+    SYSTEM="64 bit (${ARCH})"
+    ;;
+  i686|i386 )
+    SYSTEM="32 bit (${ARCH})"
+    ;;
+  *[Aa][Rr][Mm]* )
+    SYSTEM="ARM (${ARCH})"
+    ;;
+  * )
+    echo "Error: unrecognised architecture '${ARCH}'"
+    exit 1
+    ;;
+esac
+
+if [ "${APPNAME}" == "MuseScore" ]; then
+  # Upload a new version but don't publish it (invisible until published)
+  url_query="" # Don't publish, don't overwrite existing files with same name
+  PCK_NAME="$APPNAME-$ARCH"
+  BINTRAY_REPO="test" # "nightlies-linux"
```

Comment:
@lasconic I just noticed you already created a repo for the actual releases so I'll update this later today.